addons/addon-base-raas/packages/base-raas-services/lib/environment/service-catalog/environment-sc-cidr-service.js

```diff
@@ -0,0 +1,221 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const _ = require('lodash');
+const IsCidr = require('is-cidr');
+const Service = require('@aws-ee/base-services-container/lib/service');
+
+class EnvironmentScCidrService extends Service {
+  constructor() {
+    super();
+    this.dependency([
+      'aws',
+      'environmentScService',
+      'environmentAuthzService',
+      'auditWriterService',
+      'authorizationService',
+      'lockService',
+    ]);
+  }
+
+  async init() {
+    await super.init();
+    const environmentAuthzService = await this.service('environmentAuthzService');
+    // A private authorization condition function that just delegates to the environmentAuthzService
+    this._allowAuthorized = (requestContext, { resource, action, effect, reason }, ...args) =>
+      environmentAuthzService.authorize(requestContext, { resource, action, effect, reason }, ...args);
+  }
+
+  checkRequest(updateRequest) {
+    if (_.isUndefined(updateRequest) || _.isEmpty(updateRequest))
+      throw this.boom.badRequest('The request made had an empty body. Please check your payload', true);
+
+    const erroneousInputs = [];
+    const ipv6Format = [];
+    _.map(updateRequest, rule => {
+      _.forEach(rule.cidrBlocks, cidrBlock => {
+        if (IsCidr(cidrBlock) === 0) erroneousInputs.push(cidrBlock);
+        if (IsCidr.v6(cidrBlock)) ipv6Format.push(cidrBlock);
+      });
+    });
+
+    if (!_.isEmpty(erroneousInputs))
+      throw this.boom.badRequest(`The following are invalid CIDR inputs: ${erroneousInputs.join(', ')}`, true);
+
+    if (!_.isEmpty(ipv6Format))
+      throw this.boom.badRequest(
+        `Currently only IPv4 formatted CIDR inputs are allowed. Please remove : ${ipv6Format.join(
+          ', ',
+        )} CIDR ranges from your list`,
+        true,
+      );
+  }
+
+  async update(requestContext, { id, updateRequest }) {
+    // Before anything, check if the payload is valid
+    this.checkRequest(updateRequest);
+
+    const [environmentScService, lockService] = await this.service(['environmentScService', 'lockService']);
+    const existingEnvironment = await environmentScService.mustFind(requestContext, { id });
+
+    // Check if user is allowed to update cidrs
+    await this.assertAuthorized(
```

Comment:
Does this also check if the user is allowed, ie owner of the sec group?

Reply:
Yes, this check passes only when the user is the owner of the workspace or an application admin.